### PR TITLE
openexr: add 2.4.3, 2.5.8 and 3.1.9

### PR DESCRIPTION
**Pull request by aumuell**
2.5.8 is the latest version compatible with OpenScenGraph

### CODE REVIEW

**Comment by aumuell** [comment]
@spackbot fix style

**Review by tldahlgren — COMMENTED** [review]
Confirmed the version sha256.

Asking @adamjstewart if he wants to comment since he made significant changes a few weeks ago.

File: var/spack/repos/builtin/packages/openexr/package.py [hunk 1]
```diff
@@ -64,12 +67,20 @@ class Openexr(CMakePackage):
...
     depends_on("ilmbase", when="@:2")
     depends_on("zlib")
 
+    # fix build of 2.5 with GCC 13
+    patch(
+        "https://github.com/AcademySoftwareFoundation/openexr/pull/1499/commits/8bb802a4ea1c9628e6b77a4bfa9e6ec9bb97ca05.patch?full_index=1",
+        sha256="5919bfeb3f87acc455373a774d6b24a6db6c089f09e6ff1821d46cd1491dbdbf",
+        when="@2.5",
+    )
+    conflicts("@:2.4 %gcc@13")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/openexr/package.py]:
```suggestion
    conflicts("@:2.4 %gcc@13:")
```
Presumably newer versions will have the same issue?

File: var/spack/repos/builtin/packages/openexr/package.py [hunk 2]
```diff
@@ -87,6 +98,6 @@ def configure_args(self):
 
         return args
 
-    @when("@:2")
+    @when("@:2.3")
     def cmake(self, spec, prefix):
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/openexr/package.py:103]: This should really be a multi-build system package. `gdal` is my go-to example of this but there are many more.

**Comment by adamjstewart** [comment]
> Asking @adamjstewart if he wants to comment since he made significant changes a few weeks ago.

Over a year ago 😆

**Comment by tldahlgren** [comment]
> > Asking @adamjstewart if he wants to comment since he made significant changes a few weeks ago.
> 
> Over a year ago 😆

What?! Must've misread `blame`:

```
$ spack blame openexr
LAST_COMMIT  LINES  %      AUTHOR            EMAIL
8 hours ago  4      4.7    aumuell           <aumuell@users.noreply.github.com>
4 days ago   8      9.4    Martin Aumüller  <aumuell@reserv.at>
3 weeks ago  26     30.6   Adam J. Stewart   <ajstewart426@gmail.com>
a month ago  1      1.2    Alec Scott        <hi@alecbcs.com>
a year ago   1      1.2    Tom Scogland      <tscogland@llnl.gov>
a year ago   1      1.2    Vanessasaurus     <814322+vsoch@users.noreply.github.com>
2 years ago  1      1.2    ketsubouchi       <kenta.tsubouchi@hac-inc.co.jp>
3 years ago  24     28.2   Todd Gamblin      <tgamblin@llnl.gov>
4 years ago  17     20.0   Phil Tooley       <32297355+ptooley@users.noreply.github.com>
7 years ago  2      2.4    mwilliammyers     <mwilliammyers@gmail.com>
                                             
8 hours ago  85     100.0                  
```

**Comment by tldahlgren** [comment]
> > > Asking @adamjstewart if he wants to comment since he made significant changes a few weeks ago.
> > 
> > 
> > Over a year ago 😆
> 
> What?! Must've misread `blame`:
> 
> ```
> $ spack blame openexr
> LAST_COMMIT  LINES  %      AUTHOR            EMAIL
> 8 hours ago  4      4.7    aumuell           <aumuell@users.noreply.github.com>
> 4 days ago   8      9.4    Martin Aumüller  <aumuell@reserv.at>
> 3 weeks ago  26     30.6   Adam J. Stewart   <ajstewart426@gmail.com>
> a month ago  1      1.2    Alec Scott        <hi@alecbcs.com>
> a year ago   1      1.2    Tom Scogland      <tscogland@llnl.gov>
> a year ago   1      1.2    Vanessasaurus     <814322+vsoch@users.noreply.github.com>
> 2 years ago  1      1.2    ketsubouchi       <kenta.tsubouchi@hac-inc.co.jp>
> 3 years ago  24     28.2   Todd Gamblin      <tgamblin@llnl.gov>
> 4 years ago  17     20.0   Phil Tooley       <32297355+ptooley@users.noreply.github.com>
> 7 years ago  2      2.4    mwilliammyers     <mwilliammyers@gmail.com>
>                                              
> 8 hours ago  85     100.0                  
> ```

Looks like checking GitHub for blame IS more accurate than `spack blame` ATM.

**Comment by adamjstewart** [comment]
Ah, I made minor changes 3 wks ago and significant changes over a year ago. So we're both right 😄

**Review by adamjstewart — DISMISSED** [review]
I'm personally fine with these changes. Only reason I touched this package was because it was a possible dep of GDAL and I wanted to test some GDAL changes. So I shouldn't be considered any kind of authority on openexr.

**Comment by aumuell** [comment]
#39863 includes the important changes

**Pull request closed without merging**